template/map.jinja

```diff
@@ -9,16 +9,44 @@
 {%- import_yaml tplroot ~ "/osmap.yaml" as osmap %}
 {%- import_yaml tplroot ~ "/osfingermap.yaml" as osfingermap %}
 
-{%- set defaults = salt['grains.filter_by'](default_settings,
-    default='template',
-    merge=salt['grains.filter_by'](osfamilymap, grain='os_family',
-      merge=salt['grains.filter_by'](osmap, grain='os',
-        merge=salt['grains.filter_by'](osfingermap, grain='osfinger',
-          merge=salt['config.get']('template:lookup', default={})
+{%- if salt['config.get']('__cli') == 'salt-call' %}
+  {%- set defaults = salt['grains.filter_by'](default_settings,
+      default='template',
+      merge=salt['grains.filter_by'](osfamilymap, grain='os_family',
+        merge=salt['grains.filter_by'](osmap, grain='os',
+          merge=salt['grains.filter_by'](osfingermap, grain='osfinger',
+            merge=salt['pillar.get']('template:lookup', default={})
+          )
         )
       )
-    )
-) %}
+  ) %}
 
-{#- Merge the template config (e.g. from pillar) #}
-{%- set template = salt['config.get']('template', default=defaults, merge=True) %}
```

Comment:
AFAIR  the key in "`salt` vs. `salt-ssh`" was that `defaults.merge` does/did not work on `salt-ssh`. I suppose `config.get` works just fine, but I'd have to test that.

So if we could use `config.get` and stick to `grains.filter_by` (instead of `defaults.merge`) we'd just need _one_ code path without distinguishing between `salt` and `salt-ssh`.

From my tests `salt-ssh` seems to be able to use `config.get`:

```yaml
% salt-ssh host config.get apache:configdir
 host:
    /etc/apache2
 % salt-ssh host pillar.get apache:configdir
host:
    /etc/apache2
```

Reply:
`config.get` wasn't ever the issue itself (for `salt-ssh`) but the problem faced has been merging back the return from `config.get` into the rest of the map.  `defaults.merge` is one way of reliably achieving this but if `grains.filter_by` can be coerced do it instead, that's even better.